### PR TITLE
[CT-1469] init pr for short term python version issue fix

### PR DESCRIPTION
**Pull request by McKnight-42**
resolves https://github.com/dbt-labs/dbt-core/issues/6203

### Description
Upgrade actions/setup-python from v2 to v4.3.0 to bring in all the updates for deprecations
Defines Python Version to use when installing dependencies

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: dbt/adapters/snowflake/connections.py
Comment (leahwicz) [inline dbt/adapters/snowflake/connections.py:37]: Is this something new needed?

Comment (McKnight-42) [reply dbt/adapters/snowflake/connections.py:37]: no this will be removed, looking into the failure cause think its around some structure logging changes will fix

**Comment by McKnight-42** [comment]
@leahwicz this should now be good for review to start the snowflake changes and backports

**Review by dbeatty10 — APPROVED** [review]
This looks good to me 👍 

Found #307 when troubleshooting why a test for #280 was failing.

**Comment by dbeatty10** [comment]
@McKnight-42 do we want these backport labels like in https://github.com/dbt-labs/dbt-redshift/pull/211?
image

It doesn't look like https://github.com/dbt-labs/dbt-bigquery/pull/378 or https://github.com/dbt-labs/dbt-spark/pull/508 have the backport labels 🤷

**Comment by McKnight-42** [comment]
@dbeatty10 they will have to be backported but our gha isn't working on 1.2.latest and 1.1.latest last I checked so have to do those manually

**Comment by dbeatty10** [comment]
@McKnight-42 who will merge this PR? I am happy to do so if desired. I think our CI will continue to fail until this is merged, so I am waiting for this before merging https://github.com/dbt-labs/dbt-snowflake/pull/280.

Also: who will create the backports for 1.2.latest and 1.1.latest?

**Pull request merged**
